### PR TITLE
Fix the test issue#12902.

### PR DESCRIPTION
**Pull request by huanwu**
The test failed because in the core, when call XmlDictionaryWriter.CreateTextWriter, it will return an object of XmlDictionaryAsyncCheckWriter, which not implment the IXmlTextWriterInitializer. Update the test code as a simple basic one.

The code in the core:
```
public static XmlDictionaryWriter CreateTextWriter(Stream stream, Encoding encoding, bool ownsStream)
 {
      XmlUTF8TextWriter writer = new XmlUTF8TextWriter();
      writer.SetOutput(stream, encoding, ownsStream);
      var asyncWriter = new XmlDictionaryAsyncCheckWriter(writer);
       return asyncWriter;
}
```

The code in desktop:
```
static public XmlDictionaryWriter CreateTextWriter(Stream stream, Encoding encoding, bool ownsStream)
{
       XmlUTF8TextWriter writer = new XmlUTF8TextWriter();
       writer.SetOutput(stream, encoding, ownsStream);
       return writer;
}
```

#12902

@shmao @zhenlan @mconnew

### CODE REVIEW

File: src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs [hunk 1]
```diff
@@ -4140,3 +4140,17 @@ void IDeserializationCallback.OnDeserialization(object sender)
...
     }
 }
+
+public class SampleTextWriter : IXmlTextWriterInitializer
+{
+    public Encoding Encoding;
+    public Stream Stream;
+    public SampleTextWriter()
+    {
+    }
```
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs]: Nit: add an empty line below '{'

File: src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs [hunk 2]
```diff
@@ -4140,3 +4140,17 @@ void IDeserializationCallback.OnDeserialization(object sender)
         _area = Edge * Edge;
     }
 }
+
+public class SampleTextWriter : IXmlTextWriterInitializer
```
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs]: Nit: SampleTextWriter -> SampleXmlTextWriterInitializer

File: src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs [hunk 3]
```diff
@@ -278,26 +278,13 @@ public static void IXmlBinaryReaderWriterInitializerTest()
         binaryReader.Close();
     }
 
-    [ActiveIssue(12902)]
     [Fact]
     public static void IXmlTextReaderInitializerTest()
     {
-        DataContractSerializer serializer = new DataContractSerializer(typeof(TestData));
+        SampleTextWriter writer = new SampleTextWriter();
```
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs]: Nit: use 'var'

File: src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs [hunk 4]
```diff
@@ -278,26 +278,13 @@ public static void IXmlBinaryReaderWriterInitializerTest()
...
         MemoryStream ms = new MemoryStream();
```
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs]: Nit: use 'var'

**Comment by karelz** [comment]
@huanwu can you please use descriptive titles? Especially in checkin comments.
For example: Fix IXmlTextReaderInitializerTest

**Pull request merged**